From: Matthias Schiffer <mschiffer@universe-factory.net>
Date: Sat, 15 Oct 2016 19:10:33 +0200
Subject: ar71xx: set EU region code for TP-Link TL-WA901ND v4

There is no US firmware for the TL-WA901ND v4 yet, so we'll just
unconditionally set the EU region for now.

This makes LEDE flashable on these devices again. The format of the region
string is slightly different from the one used on the Archer C7 that is
generated by mktplinkfw (the second half of the region string is missing),
but it's similar enough to make it work.

Tested-by: Jannis Pinter <jannis@pinterjann.is>
Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>

Backport of LEDE 83f7ec31f8f9641ef842212c45db61b72682debf

diff --git a/target/linux/ar71xx/image/Makefile b/target/linux/ar71xx/image/Makefile
index 95f6d46..9ac1d60 100644
--- a/target/linux/ar71xx/image/Makefile
+++ b/target/linux/ar71xx/image/Makefile
@@ -736,6 +736,7 @@ define Device/tl-wa901nd-v4
     BOARDNAME := TL-WA901ND-v4
     DEVICE_PROFILE := TLWA901
     TPLINK_HWID := 0x09010004
+    IMAGE/factory.bin := append-rootfs | mktplinkfw factory -C EU
 endef
 
 TARGET_DEVICES += tl-wa901nd-v3 tl-wa901nd-v4
